cpp/src/parquet/column_reader.cc

```diff
@@ -50,6 +51,140 @@ using arrow::internal::checked_cast;
 
 namespace parquet {
 
+namespace {
+
+inline void CheckLevelRange(const int16_t* levels, int64_t num_levels,
```

Comment:
I think it would be nice to move the low-level handling of levels to a dedicated C++ file.